### PR TITLE
[no GBP] Fixes my improper skateboard fix

### PR DESCRIPTION
**Pull request by Ebb-Real**
## About The Pull Request

in #65313 i fixed skateboards being able to fly but i should have coded it differently because its still possible to fly by just spamming ollie and not moving. This does it in a more proper way.

## Why It's Good For The Game

fixes my bad fix.

## Changelog

:cl:
fix: You can no longer fly by spamming ollie onto an empty space and not moving.
/:cl:

### CODE REVIEW

File: code/modules/vehicles/vehicle_actions.dm [hunk 1]
```diff
@@ -306,24 +306,23 @@
...
-		if((locate(/obj/structure/table) in vehicle.loc.contents) || (locate(/obj/structure/fluff/tram_rail) in vehicle.loc.contents))
+		if((locate(/obj/structure/table) in landing_turf) || (locate(/obj/structure/fluff/tram_rail) in landing_turf))
 			if(locate(/obj/structure/fluff/tram_rail) in vehicle.loc.contents)
 				rider.client.give_award(/datum/award/achievement/misc/tram_surfer, rider)
 			vehicle.grinding = TRUE
 			vehicle.icon_state = "[initial(vehicle.icon_state)]-grind"
 			addtimer(CALLBACK(vehicle, /obj/vehicle/ridden/scooter/skateboard/.proc/grind), 2)
 		else
 			vehicle.obj_flags &= ~BLOCK_Z_OUT_DOWN
+		rider.spin(4, 1)
```
Comment (LemonInTheDark) [inline code/modules/vehicles/vehicle_actions.dm:317]: Doesn't this mean you'll still do the ollie animation if you run out of stanima mid go?

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]
could you describe what was wrong about your previous implementation, and then how you fixed it? 
It makes reviewing this stuff a lot easier, since I just need to check that it would work, instead of trying to figure out what you were tryin to do

**Comment by Ebb-Real** [comment]
> could you describe what was wrong about your previous implementation, and then how you fixed it? It makes reviewing this stuff a lot easier, since I just need to check that it would work, instead of trying to figure out what you were tryin to do

yeah shoulda explained that from the start, sorry bout that.

The `BLOCK_Z_OUT_DOWN` gets removed only after the code that makes the vehicle do the ollie has run. This means it will still be immune to falling while moving by the ollie action button. The changes i made makes it only remove the `BLOCK_Z_OUT_DOWN` flag if it doesn't find something to grind on, so that it won't be immune to falling when moving with ollie.

File: code/modules/vehicles/vehicle_actions.dm [hunk 2]
```diff
@@ -307,6 +307,14 @@
 			rider.Paralyze(40)
 			vehicle.visible_message(span_danger("[rider] misses the landing and falls on [rider.p_their()] face!"))
 		else
```
Comment (LemonInTheDark) [inline code/modules/vehicles/vehicle_actions.dm:309]: A suggestion:
Rather then using an else here, and indenting further right, making things harder to read, why not early return? https://github.com/tgstation/tgstation/blob/master/.github/guides/STYLE.md#use-early-returns

The same can actually be said of the if(.) at the top of this proc, I bet it'd make things a bit nicer to read

**Comment by LemonInTheDark** [comment]
> > could you describe what was wrong about your previous implementation, and then how you fixed it? It makes reviewing this stuff a lot easier, since I just need to check that it would work, instead of trying to figure out what you were tryin to do
> 
> yeah shoulda explained that from the start, sorry bout that.
> 
> The `BLOCK_Z_OUT_DOWN` gets removed only after the code that makes the vehicle do the ollie has run. This means it will still be immune to falling while moving with the ollie action button. The changes i made makes it remove the `BLOCK_Z_OUT_DOWN` flag if it doesn't find something to grind on, so that it won't be immune to falling when moving with ollie. This now gets done before the ollie code runs so that it won't have the flag while moving.

Appreciate it, thank you, I see why you're doing what you are here

**Comment by Ebb-Real** [comment]
yeah early returns are good, i think i did it right.

**Pull request merged**